app-misc/calamares-config-livecd/files/modules/downloadstage3/main.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+import os
+import shutil
+import urllib.request
+import tarfile
+import subprocess
+import libcalamares
+import glob
+
+def _progress_hook(count, block_size, total_size):
+    percent = int(count * block_size * 100 / total_size)
+    if percent > 100:
+        percent = 100
+    libcalamares.job.setprogress(percent / 2)
+
+def run():
+    url = "https://distfiles.gentoo.org/releases/amd64/autobuilds/20250420T121009Z/stage3-amd64-openrc-20250420T121009Z.tar.xz"
```

Comment:
This should verify the signature of the file.

Reply:
> This should verify the signature of the file.

noted. good point.